### PR TITLE
Rodentia Species

### PR DESCRIPTION
**Pull request by portfiend**
## About the PR
adds Rodentia, a new species of anthropomorphic rodents (with some non-rodent-but-still-adjacent marking options). 

## Why / Balance
rodents, rabbits, bats, possums are all species that are currently not covered by our customization system but have some level of demand to them. i made sure to give this species a lot of options customization-wise to cover all of these bases. however, someone else mentioned on Discord that bats should probably also be their own species too (these rodentia "bats" are wingless) and i generally agree with that sentiment

## Technical details
Rodentia's rummage produces garbage like 70% of the time and edible food the remaining 30%. This food is either a single wedge of cheese (not a wheel), a snack, or (rarely) a single-serving meal. Keep in mind that rummaging is ~~one-time per disposal chute~~ once per disposal bin on a cooldown - not per-rodentia - so each chute on the station can only be rummaged once every 5 minutes. This combined with competition with other hungry rodentia and limited access means that rodentia are unlikely to produce lots of food. I can still adjust the numbers though

mouth storage ability grants the following:
- a 2x2 storage in their mouth, which cannot store sharp objects or animals
- being pushed, attacked for more than 2 damage, or knocked over spits out the items
- other people can tell if the rodentia has something in their mouth
- they cannot eat or speak intelligibly if something is in their mouth

the name scheme of Rodentia is "Last First", where Last is a descriptor (strong, big-eyed, trash-eating) and First is either a) a pet name (nibbles, peanut) b) a simple "first name" (monty, basil), or c) what sounds like the name of an english furry-themed town (Squeakson)

## Media

Screenshot_149
Screenshot_147

Screenshot_144 Screenshot_143 Screenshot_142 Screenshot_141

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
Rummage action on disposal chutes is now bound to the Rummager component rather than the RatKing component. If you want to give an entity the ability to rummage through trash, give them "Rummager"

RatKingRummageable objects now use a 5-minute cooldown instead of a one-time "Looted" flag. setting `RatKingRummageable.Looted = true` will still disable the object from further rummaging, but now when the object is looted the `RatKingRummageable.LastLooted` property is set instead

adds a new Naming scheme type, LastFirst, which is added to the switch statement in NamingSystem.cs

mousetraps can now be always activated with a new component, AlwaysTriggerMousetrapComponent

**Changelog**

:cl: portfiend
- add: Added Rodentia, a new species of playable rodent-like humanoids.
- tweak: Mobs that are capable of rummaging (such as the new Rodentia species) can do so every five minutes. They will find small snacks in trash bins instead of entire wheels of cheese.

## Other Notes
if you're from another server, yes you can port this

most of the discussion on this species is taking place on the DV suggestions forum in the discord but i will read replies here too

# THIS PR IS MIT-LICENSED, HAS BEEN SINCE SEPTEMBER 2024: https://github.com/DeltaV-Station/Delta-v/pull/1511#issuecomment-2339581002

### CODE REVIEW

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]
Isn't this just a Felinid with some extra markings?

**Comment by portfiend** [comment]
> Isn't this just a Felinid with some extra markings?

yeah, i have said in multiple places that the playstyle of this is not unique enough for my tastes which is why i have been requesting suggestions here and on discord

**Comment by dvir001** [comment]
Going to the chef be like
[image](https://github.com/user-attachments/assets/bb4f8b86-c580-4717-b3cf-e54c0d2636aa)

Can probably give it an extra faction shared with the rat king and rats.

File: Content.Server/DeltaV/Abilities/CrawlUnderObjectsSystem.cs
Comment (DEATHB4DEFEAT) [inline Content.Server/DeltaV/Abilities/CrawlUnderObjectsSystem.cs]:
```suggestion
        SubscribeLocalEvent<CrawlUnderObjectsComponent, RefreshMovementSpeedModifiersEvent>(OnRefreshMovespeed);
```

File: Content.Server/DeltaV/Abilities/CrawlUnderObjectsSystem.cs
Comment (DEATHB4DEFEAT) [inline Content.Server/DeltaV/Abilities/CrawlUnderObjectsSystem.cs]:
```suggestion
        if (component.Enabled
        	|| (TryComp<ClimbingComponent>(uid, out var climbing)
        		&& climbing.IsClimbing == true))
            return false;
```

File: Content.Server/DeltaV/Abilities/CrawlUnderObjectsSystem.cs
Comment (DEATHB4DEFEAT) [inline Content.Server/DeltaV/Abilities/CrawlUnderObjectsSystem.cs]:
```suggestion
        if (!component.Enabled
        	|| IsOnCollidingTile(uid))
        	|| (TryComp<ClimbingComponent>(uid, out var climbing)
        		&& climbing.IsClimbing == true))
            return false;
```

File: Content.Server/DeltaV/Abilities/CrawlUnderObjectsSystem.cs
Comment (DEATHB4DEFEAT) [inline Content.Server/DeltaV/Abilities/CrawlUnderObjectsSystem.cs]:
```suggestion
        if (TryComp(uid, out FixturesComponent? fixtureComponent))
            foreach (var (key, originalMask) in component.ChangedFixtures)
                if (fixtureComponent.Fixtures.TryGetValue(key, out var fixture))
                    _physics.SetCollisionMask(uid, key, fixture, originalMask, fixtureComponent);

```

File: Content.Shared/DeltaV/Abilities/AlwaysTriggerMousetrapComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Abilities/AlwaysTriggerMousetrapComponent.cs]:
```suggestion
public sealed partial class AlwaysTriggerMousetrapComponent : Component { }
```

File: Content.Shared/DeltaV/Abilities/CrawlUnderObjectsComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Abilities/CrawlUnderObjectsComponent.cs]:
```suggestion
public sealed partial class ToggleCrawlingStateEvent : InstantActionEvent { }
```

File: Content.Shared/DeltaV/Abilities/RummagerComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Abilities/RummagerComponent.cs]:
```suggestion
public sealed partial class RummagerComponent : Component { }
```

File: Content.Shared/DeltaV/Storage/Components/MouthStorageComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Storage/Components/MouthStorageComponent.cs]:
```suggestion
    [DataField]
```

File: Content.Shared/DeltaV/Storage/Components/MouthStorageComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Storage/Components/MouthStorageComponent.cs]:
```suggestion
    [DataField]
```

File: Content.Shared/DeltaV/Storage/Components/MouthStorageComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Storage/Components/MouthStorageComponent.cs]:
```suggestion
    [DataField, AutoNetworkedField]
```

File: Content.Shared/DeltaV/Storage/Components/MouthStorageComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Storage/Components/MouthStorageComponent.cs]:
```suggestion
    [DataField]
```

File: Content.Shared/DeltaV/Storage/EntitySystems/SharedMouthStorageSystem.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Storage/EntitySystems/SharedMouthStorageSystem.cs]:
```suggestion
        if (!string.IsNullOrWhiteSpace(component.OpenStorageAction) && component.Action == null)
            _actionsSystem.AddAction(uid, ref component.Action, component.OpenStorageAction, mouth);
```

File: Content.Shared/Humanoid/NamingSystem.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/Humanoid/NamingSystem.cs:45]: Shouldn't you just swap where you put the name lists?

File: Resources/Locale/en-US/deltav/actions/crawl-under-objects.ftl
Comment (DEATHB4DEFEAT) [inline Resources/Locale/en-US/deltav/actions/crawl-under-objects.ftl]:
```suggestion
crawl-under-objects-toggle-on = Now sneaking
crawl-under-objects-toggle-off = Now standing
```

File: Resources/Prototypes/DeltaV/Voice/speech_emote_sounds.yml
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/DeltaV/Voice/speech_emote_sounds.yml]:
```suggestion
      path: /Audio/Animals/mouse_squeak.ogg

```

File: Resources/Prototypes/Nyanotrasen/metempsychoticHumanoids.yml
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/Nyanotrasen/metempsychoticHumanoids.yml]:
```suggestion
    Rodentia: 0.5

```

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]

File: Content.Shared/Humanoid/NamingSystem.cs
Comment (portfiend) [reply Content.Shared/Humanoid/NamingSystem.cs:45]: first names are gendered, last names are not

File: Content.Server/DeltaV/Abilities/CrawlUnderObjectsSystem.cs
Comment (DEATHB4DEFEAT) [inline Content.Server/DeltaV/Abilities/CrawlUnderObjectsSystem.cs]:
```suggestion
```

File: Content.Server/DeltaV/Abilities/CrawlUnderObjectsSystem.cs
Comment (DEATHB4DEFEAT) [inline Content.Server/DeltaV/Abilities/CrawlUnderObjectsSystem.cs]:
```suggestion
            || (TryComp<ClimbingComponent>(uid, out var climbing)
                && climbing.IsClimbing == true))
```

File: Content.Server/DeltaV/Abilities/CrawlUnderObjectsSystem.cs
Comment (DEATHB4DEFEAT) [inline Content.Server/DeltaV/Abilities/CrawlUnderObjectsSystem.cs]:
```suggestion
            || IsOnCollidingTile(uid)
            || (TryComp<ClimbingComponent>(uid, out var climbing)
                && climbing.IsClimbing == true))
```

File: Content.Shared/DeltaV/Abilities/AlwaysTriggerMousetrapComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Abilities/AlwaysTriggerMousetrapComponent.cs]:
```suggestion
[RegisterComponent, NetworkedComponent]
```

File: Content.Shared/DeltaV/Abilities/AlwaysTriggerMousetrapComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Abilities/AlwaysTriggerMousetrapComponent.cs]:
```suggestion
public sealed partial class AlwaysTriggerMousetrapComponent : Component;
```

File: Content.Shared/DeltaV/Abilities/CrawlUnderObjectsComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Abilities/CrawlUnderObjectsComponent.cs]:
```suggestion
[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
```

File: Content.Shared/DeltaV/Abilities/CrawlUnderObjectsComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Abilities/CrawlUnderObjectsComponent.cs]:
```suggestion
    [DataField]
```

File: Content.Shared/DeltaV/Abilities/CrawlUnderObjectsComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Abilities/CrawlUnderObjectsComponent.cs]:
```suggestion
[Serializable, NetSerializable]
public sealed partial class CrawlingUpdatedEvent(bool enabled = false) : EventArgs
{
    public readonly bool Enabled = enabled;
}
```

File: Content.Shared/DeltaV/Abilities/RummagerComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Abilities/RummagerComponent.cs]:
```suggestion
[RegisterComponent, NetworkedComponent]
public sealed partial class RummagerComponent : Component;
```

File: Content.Shared/DeltaV/Storage/Components/MouthStorageComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Storage/Components/MouthStorageComponent.cs]:
```suggestion
[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
```

File: Content.Shared/DeltaV/Storage/EntitySystems/SharedMouthStorageSystem.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Storage/EntitySystems/SharedMouthStorageSystem.cs:23]:
```suggestion
        base.Initialize();

```

File: Content.Shared/DeltaV/Storage/EntitySystems/SharedMouthStorageSystem.cs
Comment (DEATHB4DEFEAT) [inline Content.Shared/DeltaV/Storage/EntitySystems/SharedMouthStorageSystem.cs]:
```suggestion
        if (args.DamageDelta == null
            || !args.DamageIncreased
            || args.DamageDelta.GetTotal() < component.SpitDamageThreshold)
```

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]

File: Resources/Locale/en-US/deltav/markings/rodentia.ftl
Comment (DEATHB4DEFEAT) [inline Resources/Locale/en-US/deltav/markings/rodentia.ftl:1]: I think single `#`s are technically invalid Fluent, but RT doesn't care

**Review by DEATHB4DEFEAT — APPROVED** [review]
I'll let Deltanedas or someone decide on if they want this/features of this or whatever, but the code looks fine.

**Comment by portfiend** [comment]
ready for merge when maintainers approve it

File: Content.Server/DeltaV/Storage/EntitySystems/MouthStorageSystem.cs
Comment (deltanedas) [inline Content.Server/DeltaV/Storage/EntitySystems/MouthStorageSystem.cs]: 1. use EntitySystem's EntityManager field instead of resolving it
2. move this to shared

File: Content.Server/Mousetrap/MousetrapSystem.cs
Comment (deltanedas) [inline Content.Server/Mousetrap/MousetrapSystem.cs]: use HasComp

File: Content.Shared/Storage/Components/DumpableComponent.cs
Comment (deltanedas) [inline Content.Shared/Storage/Components/DumpableComponent.cs]: remove this event and just have the method be public api on dumpable system

File: Content.Shared/Storage/EntitySystems/DumpableSystem.cs
Comment (deltanedas) [inline Content.Shared/Storage/EntitySystems/DumpableSystem.cs]: make it public and have component be optional and resolved
instead of raising meme event `_dumpable.DumpContents(uid, null, null, user);`

File: Resources/Prototypes/DeltaV/Body/Parts/rodentia.yml
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Body/Parts/rodentia.yml]: remove all netsync falses, put the sprite+icon paths on PartRodentia

File: Resources/Prototypes/DeltaV/Entities/Mobs/Player/rodentia.yml
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Mobs/Player/rodentia.yml:1]: this and BaseMobRodentia yml have huge duplication with BaseSpeciesOrganic, remove everything thats inherited

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Review by deltanedas — APPROVED** [review]
code looks good now but needs lyndos approval too

**Comment by Lyndomen** [comment]
ill get back to you in 3 business days

**Comment by Verbalase** [comment]
> ill get back to you in 3 business days

what did bro really mean by this

**Review by Lyndomen — APPROVED** [review]
game director  approved

**Comment by ghost** [comment]
> > ill get back to you in 3 business days
> 
> what did bro really mean by this

watch this you goober!!

**Comment by ghost** [comment]
THE MICE ARE COMING!!

**Comment by jimmy12or** [comment]
Yay I can make my bun!

**Comment by portfiend** [comment]
hi also when i wrote "forks are free to port and use this" i really did mean it, which is to say if merely me giving the word is enough to will it into reality then the additions and code enclosed in this PR are licensed under the MIT license thanks

**Pull request merged**
